src/js/stores/library.js

```diff
@@ -214,6 +253,144 @@ define(function (require, exports, module) {
         },
         
         /**
+         * Handle open-graphic-for-edit event by creating an edit status for the opened document.
+         *
+         * @private
+         * @param {object} payload
+         * @param {number} payload.documentID
+         * @param {AdobeLibraryElement} payload.element
+         * @param {string} payload.documentPath
+         * @param {string} payload.previewPath
+         */
+        _handleOpenGraphicForEdit: function (payload) {
+            var documentID = payload.documentID,
+                element = payload.element,
+                documentPath = payload.documentPath,
+                previewPath = payload.previewPath;
+            
+            this._graphicEditStatusByDocumentID[documentID] = {
+                documentPath: documentPath,
+                previewPath: previewPath,
+                element: element,
+                isUpdatingContent: false,
+                isDocumentClosed: false
+            };
+        },
+        
+        /**
+         * Handle save document. If an element is associated with the document, we update the element's 
+         * content and preview image.
+         *
+         * @private
+         * @param {object} payload
+         * @param {number} payload.documentID 
+         * @param {string} payload.path - document path
+         */
+        _handleSaveDocument: function (payload) {
+            var documentID = payload.documentID,
+                editStatus = this._graphicEditStatusByDocumentID[documentID] ||
+                    this._rebuildEditStatus(documentID, payload.path);
+                
+            if (!editStatus) {
+                return;
+            }
+            
+            var element = editStatus.element,
+                documentPath = editStatus.documentPath,
+                previewPath = editStatus.previewPath;
+            
+            editStatus.isUpdatingContent = true;
+            
+            this.flux.actions.libraries.updateGraphicContent(element, documentPath, previewPath)
```

Comment:
Enqueueing actions from stores are a bad idea, can you transfer to this from the save action somehow? Paging @iwehrman for his opinion/idea as well.